client/src/components/BarChart.js

```diff
@@ -204,44 +168,59 @@ class BarChart extends Component {
       .attr("height", function(d) {
         return yHeight - yScale(getPropValue(d, yProp))
       })
-      .on(
-        "mouseenter",
-        d =>
-          this.props.showPopover && this.props.showPopover(d3.event.target, d)
-      )
-      .on("mouseleave", d => this.props.hidePopover && this.props.hidePopover())
+      .attr("data-for", "tooltip-top")
+      .attr("data-html", true)
+      .attr("data-tip", d => tooltip && tooltip(d))
     if (onBarClick) {
       bar.on("click", function(d) {
         onBarClick(d)
       })
     }
-  }
 
-  render() {
-    return (
-      <svg
-        id={this.props.chartId}
-        ref={this.node}
-        width={this.props.width}
-        height={this.props.height}
-      />
-    )
+    ReactTooltip.rebuild()
+  }, [
+    node,
+    width,
+    height,
+    chartId,
+    data,
+    xProp,
+    yProp,
+    xLabel,
+    barClass,
+    onBarClick,
+    tooltip,
+    selectedBarClass,
+    selectedBar
+  ])
+
+  return <svg id={chartId} ref={node} width={width} height={height} />
+
+  function isNumeric(value) {
```

Comment:
isNumeric could be placed outside the BarChart component.

Reply:
Same here.